### ISSUE TITLE
What if global package cache will be deleted?

### ISSUE THREAD

**Issue by CorradoLanera** [rstudio/renv#90]
Suppose I used renv for some projects tracked on GitHub, I will work on them for some times so that they all use different versions of different packages. Great. 

Now, I have to restore my laptop, erasing everything (and I forgot to backup my renv global package cache). I can quickly restore all my projects, with all their history and the corresponding lock files, but I do not have my global package cache anymore.

What happens in this situation? What/how can I do to restore the global package cache? My projects still work (and with which packages versions)?

**Comment by kevinushey** [comment]
Short version: as long as the packages as specified in the lockfile can still be recovered, they can be reinstalled with `renv::restore()`. That action (`renv::restore()`) will re-populate the cache as packages are reinstalled.

---

Longer version, here's what I see in a project using `digest` if I've removed the cache:

```
kevin@cdrv:~/dummyproj
$ R

R version 3.6.0 (2019-04-26) -- "Planting of a Tree"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin15.6.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

* Project '~/dummyproj' loaded. [renv 0.5.0-17]
Warning message:
The project library's symlinks to the cache are all broken.
Has the cache been removed, or is it otherwise inaccessible?
Cache root: '/Users/kevin/Library/Application Support/renv/cache/v4/R-3.6/x86_64-apple-darwin15.6.0'
> renv::status()
The following package(s) have broken symlinks into the cache:

        digest

Consider re-installing these packages.

The following package(s) are recorded in the lockfile but not installed:
         _
  digest   [0.6.19]

Use `renv::restore(actions = "install")` to install these packages.
```

Following the instructions:

```
> renv::restore()
The following package(s) have broken symlinks into the cache:

        digest

Consider re-installing these packages.

The following package(s) will be installed:
         _
  digest   [0.6.19]

Do you want to proceed? [y/N]: y
* Querying repositories for available binary packages ... Done!
Retrieving 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.6/digest_0.6.19.tgz' ...
        OK [downloaded 210 Kb in 0.2 secs]
Installing digest [0.6.19] from CRAN ...
        OK (installed binary)
> renv::status()
* The project is already synchronized with the lockfile.
```

Inspecting the cache:

```
> renv:::renv_cache_list()
[1] "/Users/kevin/Library/Application Support/renv/cache/v4/R-3.6/x86_64-apple-darwin15.6.0/digest/0.6.19/2e0bb891e01c4e3f1132304cf5dd4aaa/digest"
```

So, the cache will be recovered and rebuilt as your projects are restored. Does that make sense?

**Comment by kevinushey** [comment]
That said, one needs to be aware that:

1. Some packages can be challenging to install from sources; if you've just restored a machine it might be challenging to e.g. recover the system libraries or other dependencies that a package needed during installation. Currently, `renv` doesn't do anything to help with this, but in the future https://github.com/r-hub/sysreqsdb will help.

2. If your project depends on R packages which are either not available from an external source (e.g. you installed the package from a local tarball) then the package may not be recoverable.

In other words, if you decide to have mission-critical applications that depend on the `renv` cache, it would be prudent to consider backing up the cache explicitly. I believe the same story would be true for e.g. systems hosting Shiny applications with R packages installed into a site or user library, for example.

**Comment by CorradoLanera** [comment]
Thank you, the solution is clear.
